### DISCUSSION TITLE
feature: Add the ability to control the packages cache duration

### DISCUSSION THREAD

**Discussion by nabeelsaabna** [renovatebot/renovate#23806] [Suggest an Idea]
### Tell us more.

In below functions we see that the values are hardcoded for `ttlMinutes: 60` or it's not mentioned and falls back to the default value of `30`

https://github.com/renovatebot/renovate/blob/4db86c202a92953f1db2909a536b8888afe722d9/lib/modules/datasource/docker/index.ts#L382-L392

https://github.com/renovatebot/renovate/blob/4db86c202a92953f1db2909a536b8888afe722d9/lib/modules/datasource/docker/index.ts#L596-L604

I want the ability to control the cache duration from external configurations by `datasource` and cache `namespace`.

Possible solution is to add a config option: (hierarchy of `datasource` added for an easier implementation)

```
{
	"cacheTTL": {
		"docker": {
			"datasource-docker-labels": 150 
			"datasource-docker-tags": 200
		}
	}
}
```

WDYT ?

**Comment by nabeelsaabna** [comment]
maybe ?
```
{
	"cacheTTL": {
		"docker/datasource-docker-labels": 150,
		"docker/datasource-docker-tags": 200
	}
}
```

**Reply by rarkins** [reply]
I prefer a flattened approach like this
